### ISSUE TITLE
Create snapshots by cron 

### ISSUE THREAD

**Issue by knazarov** [tarantool/tarantool#4261]
In some cases when there are many tarantool instances on a machine, synchronous snapshotting may occur, which can significantly degrade quality of service.

In order to solve this, I propose to take this library: https://github.com/tarantool/cron-parser and add an option to box.cfg to produce snapshots by a cron expression.

**Comment by kostja** [comment]
See also: https://github.com/tarantool/tarantool/issues/1937 and https://github.com/tarantool/tarantool/issues/466
We need to have a common process shared by all tarantool instances on the machine. It should work as a global quota/resource manager for all shared resources.

**Comment by Totktonada** [comment]
Probably we should have a subcommand in tarantoolctl that sends SIGUSR1 to a corresponsing tarantool instance (or connects and evals `box.snapshot()`). After that it is easy to add this command into a crontab file. But built-in support for that prorably would simplify an app configuration: it will not necessarily spread across files.

**Comment by kyukhin** [comment]
Taking last 2 comments into account, looks like we don't need to have cron-based timetable onboard. We might need to add a command to `tarantoolctl`  which implements snapshotting of a given instance. In that cases this manager turns into simple bash script which is invoked by system-wide cron.

**Comment by kostja** [comment]
This would  spread tarantool configuration in multiple places. Plus users will have to manually spread checkpoints of different instances apart. @Kasen has already done cron string parsing btw.

**Comment by knazarov** [comment]
I agree that having the user manually pick snapshot times for a cluster may be suboptimal. I like @kostja idea more: a global entity that manages this in a cluster will be nice.

**Comment by Totktonada** [comment]
The problem that is declared by all three issues (this, #1937 and #466) is about postponing snapshotting when another instance performs a snapshot and/or at peak load. I agree that it should be done automatically rather than configured manually.

There are proposals how to implement it:

* Schedule snapshots using a coordinator process.
* Synchronize tarantool instances without a coordinator process.

I like the second, because it is simpler from a user/operation perspective.

There are proposals regarding the configuration API:

* Don't extend the API, just postpone a snapshot if another instance performs it.
* Support cron-like expressions in `checkpoint_interval`.

The second means some manual per-instance configuration, so the first one seems better from a user perspective.

----

If we'll look at the problem in a bit more general way, what is the problem? Snapshot writes a lot of data that may degrade performance of other operations. It is unclear, whether it assumes other *disc* operations or a system performance at whole (a kind of https://bugzilla.kernel.org/show_bug.cgi?id=12309). I'll assume the former.

So can't we just prioritize those writes as low priority ones?
